hibernate-testing/src/main/java/org/hibernate/testing/junit4/BaseNonConfigCoreFunctionalTestCase.java

```diff
@@ -450,6 +451,11 @@ protected void prepareTest() throws Exception {
 
 	@After
 	public final void afterTest() throws Exception {
+		// see https://github.com/hibernate/hibernate-orm/pull/3412#issuecomment-678338398
+		if ( getDialect() instanceof H2Dialect ) {
+			ReflectHelper.getMethod( Class.forName( "org.h2.util.DateTimeUtils" ), "resetCalendar" ).invoke( null );
```

Comment:
Could we centralize the h2 workaround into `BaseUnitTestCase` to avoid ugly duplication?

Reply:
Also, we have other testing base classes (e.g. `BaseEntityManagerFunctionalTestCase`) but they are inherit `BaseUnitTestCase`.

Reply:
IIRC (on the phone right now), I had a look already and that `BaseUnitTestCase` does not know the current dialect.